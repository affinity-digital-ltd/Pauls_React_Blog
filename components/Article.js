import { Component } from 'react'
import gql from 'graphql-tag'

import { MarkdownPreview } from 'react-marked-markdown'
import moment from 'moment'

import UserComments from '../components/UserComments'

class Article extends Component {
  render () {
    const { article } = this.props

    return <article className='o-main-container'>
      <div className='c-article'>
        <h2 className='c-article__header'>{article.title}</h2>
        <div className='c-header-divider' />
        <h3 className='c-article__author'>Article by: {article.author}</h3>
        <time className='c-article__published'><p>Published on: {moment(article.createdAt).format('ll')}</p></time>
        <figure>
          <img className='c-article__main-img' src={`http://localhost:3001${article.image}`} />
        </figure>
        <div className='c-article__text'>
          <MarkdownPreview value={article.text} />
        </div>
      </div>
      {/* <UserComments comments={article.comments} /> */}
      <a className='c-user-comments__back-btn'>Back to Article List</a>
    </article>
  }
}

export default Article
