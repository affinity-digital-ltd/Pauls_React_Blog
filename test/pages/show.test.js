import Article from '../../pages/article'

describe('<Article />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <Article />
    )

    expect(component).toMatchSnapshot()
  })
})
