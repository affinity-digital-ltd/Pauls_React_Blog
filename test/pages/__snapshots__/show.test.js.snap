// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Article />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <article
        className="o-main-container"
      >
        <div
          className="c-article"
        >
          <h2
            className="c-article__header"
          >
            Article Title
          </h2>
          <div
            className="c-header-divider"
          />
          <h3
            className="c-article__author"
          >
            Article by Author Name
          </h3>
          <time
            className="c-article__published"
          >
            Published: Published at Date
          </time>
          <figure>
            <img
              alt="An animated rock"
              className="c-article__image"
              src="../static/images/rock.png"
            />
          </figure>
          <div
            className="c-article__text"
          >
            <p>
              Body Text
            </p>
          </div>
        </div>
        <UserComments />
        <a
          className="c-user-comments__back-btn"
        >
          Back to Article List
        </a>
      </article>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="c-article"
          >
            <h2
              className="c-article__header"
            >
              Article Title
            </h2>
            <div
              className="c-header-divider"
            />
            <h3
              className="c-article__author"
            >
              Article by Author Name
            </h3>
            <time
              className="c-article__published"
            >
              Published: Published at Date
            </time>
            <figure>
              <img
                alt="An animated rock"
                className="c-article__image"
                src="../static/images/rock.png"
              />
            </figure>
            <div
              className="c-article__text"
            >
              <p>
                Body Text
              </p>
            </div>
          </div>,
          <UserComments />,
          <a
            className="c-user-comments__back-btn"
          >
            Back to Article List
          </a>,
        ],
        "className": "o-main-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="c-article__header"
              >
                Article Title
              </h2>,
              <div
                className="c-header-divider"
              />,
              <h3
                className="c-article__author"
              >
                Article by Author Name
              </h3>,
              <time
                className="c-article__published"
              >
                Published: Published at Date
              </time>,
              <figure>
                <img
                  alt="An animated rock"
                  className="c-article__image"
                  src="../static/images/rock.png"
                />
              </figure>,
              <div
                className="c-article__text"
              >
                <p>
                  Body Text
                </p>
              </div>,
            ],
            "className": "c-article",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Article Title",
                "className": "c-article__header",
              },
              "ref": null,
              "rendered": "Article Title",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "c-header-divider",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Article by Author Name",
                "className": "c-article__author",
              },
              "ref": null,
              "rendered": "Article by Author Name",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Published: Published at Date",
                "className": "c-article__published",
              },
              "ref": null,
              "rendered": "Published: Published at Date",
              "type": "time",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="An animated rock"
                  className="c-article__image"
                  src="../static/images/rock.png"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "An animated rock",
                  "className": "c-article__image",
                  "src": "../static/images/rock.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "figure",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  Body Text
                </p>,
                "className": "c-article__text",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Body Text",
                },
                "ref": null,
                "rendered": "Body Text",
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Back to Article List",
            "className": "c-user-comments__back-btn",
          },
          "ref": null,
          "rendered": "Back to Article List",
          "type": "a",
        },
      ],
      "type": "article",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <article
          className="o-main-container"
        >
          <div
            className="c-article"
          >
            <h2
              className="c-article__header"
            >
              Article Title
            </h2>
            <div
              className="c-header-divider"
            />
            <h3
              className="c-article__author"
            >
              Article by Author Name
            </h3>
            <time
              className="c-article__published"
            >
              Published: Published at Date
            </time>
            <figure>
              <img
                alt="An animated rock"
                className="c-article__image"
                src="../static/images/rock.png"
              />
            </figure>
            <div
              className="c-article__text"
            >
              <p>
                Body Text
              </p>
            </div>
          </div>
          <UserComments />
          <a
            className="c-user-comments__back-btn"
          >
            Back to Article List
          </a>
        </article>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="c-article"
            >
              <h2
                className="c-article__header"
              >
                Article Title
              </h2>
              <div
                className="c-header-divider"
              />
              <h3
                className="c-article__author"
              >
                Article by Author Name
              </h3>
              <time
                className="c-article__published"
              >
                Published: Published at Date
              </time>
              <figure>
                <img
                  alt="An animated rock"
                  className="c-article__image"
                  src="../static/images/rock.png"
                />
              </figure>
              <div
                className="c-article__text"
              >
                <p>
                  Body Text
                </p>
              </div>
            </div>,
            <UserComments />,
            <a
              className="c-user-comments__back-btn"
            >
              Back to Article List
            </a>,
          ],
          "className": "o-main-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="c-article__header"
                >
                  Article Title
                </h2>,
                <div
                  className="c-header-divider"
                />,
                <h3
                  className="c-article__author"
                >
                  Article by Author Name
                </h3>,
                <time
                  className="c-article__published"
                >
                  Published: Published at Date
                </time>,
                <figure>
                  <img
                    alt="An animated rock"
                    className="c-article__image"
                    src="../static/images/rock.png"
                  />
                </figure>,
                <div
                  className="c-article__text"
                >
                  <p>
                    Body Text
                  </p>
                </div>,
              ],
              "className": "c-article",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Article Title",
                  "className": "c-article__header",
                },
                "ref": null,
                "rendered": "Article Title",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "c-header-divider",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Article by Author Name",
                  "className": "c-article__author",
                },
                "ref": null,
                "rendered": "Article by Author Name",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Published: Published at Date",
                  "className": "c-article__published",
                },
                "ref": null,
                "rendered": "Published: Published at Date",
                "type": "time",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="An animated rock"
                    className="c-article__image"
                    src="../static/images/rock.png"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "An animated rock",
                    "className": "c-article__image",
                    "src": "../static/images/rock.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "figure",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    Body Text
                  </p>,
                  "className": "c-article__text",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Body Text",
                  },
                  "ref": null,
                  "rendered": "Body Text",
                  "type": "p",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Back to Article List",
              "className": "c-user-comments__back-btn",
            },
            "ref": null,
            "rendered": "Back to Article List",
            "type": "a",
          },
        ],
        "type": "article",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
